data/base/endpoint.php

```diff
@@ -76,15 +78,34 @@ public function __construct( $controller ) {
 	 * @return string
 	 */
 	public function get_base_route() {
-		$endpoint_name = $this->get_name();
+		$endpoint_public_name = $this->get_command_public();
 
-		// TODO: Allow this only for internal routes.
-		// TODO: Make difference between internal and external endpoints.
-		if ( 'index' === $endpoint_name ) {
-			$endpoint_name = '';
+		if ( $endpoint_public_name ) {
+			$endpoint_public_name = '/' . $endpoint_public_name;
 		}
 
-		return '/' . $this->controller->get_rest_base() . '/' . $endpoint_name;
+		return '/' . $this->controller->get_rest_base() . $endpoint_public_name;
+	}
+
+	/**
+	 * Convert endpoint to command name ( without internal ).
+	 */
+	public function get_command_public() {
```

Comment:
By default it returns `get_name()`. on `index` override and return `/`.

Reply:
Handled.